### DISCUSSION TITLE
Inverted Airfoil from airfoil data base in Sandbox

### DISCUSSION THREAD

**Discussion by harutidmore** [peterdsharpe/AeroSandbox#104] [Q&A] [answered]
Howdy,

I am trying to run a drag optimization for a stable aircraft; I am not converging onto solutions. One reason I think this i happening is that in order to trim my current Main Wing airfoil (s1223) is very highly cambered and a symmetric airfoil would not be able to trim it. 
One solution based off preliminary sizing would be using an inverted airfoil; is there an easy way to flip the existing airfoils?
I'd prefer to avoid generating new airfoil and inverting them then importing into sandbox.

Thanks in advance
Haru

**Comment by peterdsharpe** [comment, marked as answer]
Yes, this will do it:

```python
import aerosandbox as asb
import aerosandbox.numpy as np

af = asb.Airfoil("s1223")
af_flipped = af.scale(scale_x=1, scale_y=-1)  # creates a flipped copy of `af`
af_flipped.draw()
```

If you still have convergence issues and are able to share your design script, feel free to post it here. Good luck!

**Reply by harutidmore** [reply]
Great! I'm also having an interesting bug when trying to 'draw' my plane.
I would love to be able to hop on a call on your preferred platform and we could go through my script. 
Would you be able to meet today?
here is the error:
""
..."The object type MX() has no len"...
""

I am also seeing:
"
Warning: Cutting back alpha due to evaluation error
CasADi - 2023-08-23 15:40:42 WARNING("solver:nlp_g failed: NaN detected for output g, at (row 113, col 0).") [.../casadi/core/oracle_function.cpp:326]
"
